src/lib.rs

```diff
@@ -236,3 +278,98 @@ pub fn timespecff_diff(start: &mut TimeSpec, stop: &mut TimeSpec, result: &mut T
         *result = result_sec + result_nsec;
     }
 }
+
+fn open_shmem(shm_name: &str) -> *mut c_void {
+    let shm_fd = shm_open(
+        shm_name,
+        OFlag::O_CREAT | OFlag::O_RDWR,
+        Mode::S_IRWXU | Mode::S_IRWXG | Mode::S_IRWXO,
+    )
+    .unwrap();
+    let shm_ptr = unsafe {
+        ftruncate(shm_fd, SHM_SIZE as libc::off_t);
+        mmap(
+            None,
+            NonZeroUsize::new_unchecked(SHM_SIZE),
+            ProtFlags::PROT_READ | ProtFlags::PROT_WRITE,
+            MapFlags::MAP_SHARED,
+            shm_fd,
+            0,
+        )
+        .unwrap()
+    };
+    unsafe { msync(shm_ptr, SHM_SIZE, MS_SYNC) };
+
+    shm_ptr
```

Comment:
What if shm_ptr is Null?

Reply:
mmap returned MAP_FAILED instead of null.
mmap never return null unless give MAP_FIXED flag.
FYI: https://stackoverflow.com/questions/24562691/why-does-mmap-use-map-failed-instead-of-null

Reply:
Anyway, if mmap failed, we should raise an error instead of just returning as normal.

Reply:
Please check whether returned value is valid after every unsafe system function and raise error if cannot continue to work